src/stirling/source_connectors/socket_tracer/protocols/dns/stitcher.cc

```diff
@@ -240,6 +243,96 @@ RecordsWithErrorCount<Record> StitchFrames(std::deque<Frame>* req_frames,
   return {entries, error_count};
 }
 
+// CustomStitchFrames is a re-implementation of the existing StitchFrames with an added
+// ability to record those DNS requests as well for which we could not match any response.
+// Flag include_respless_dns_requests must be set to include such DNS requests.
+RecordsWithErrorCount<Record> CustomStitchFrames(std::deque<Frame>* req_frames,
+                                                 std::deque<Frame>* resp_frames) {
```

Comment:
The high degree of similarity of this function with `PxStitchFrames` suggests we should merge them and pass in an boolean argument that selects the behavior. Otherwise there's too much duplication, which could be a problem if there's some bug fix we need to make in the future.

Also, I know you asked about this on Slack. Thanks for putting this preview out so I could get a better grasp on how much duplication there actually ends up being :)

Reply:
I'll merge the two and push the changes soon.

Reply:
fixed in a3e9243ac0b0e094651eb59e2ed6252a01080658